lib/rollup/jobs/totals_job.ex

```diff
@@ -0,0 +1,67 @@
+defmodule Castle.Rollup.Jobs.Totals do
+  # TODO: import "public" redis api from the application
+  import Castle.Redis.PartitionCache
+
+  # old - all non-today partitions from the beginning of time - 15 day cache
+  # mid - intermediate query to pick up the slack - 1 day cache
+  # new - always the current-date-partition - 5 minute cache
+  def run() do
```

Comment:
ok, so I'm totally clear -- what's happening here is: 
- every time castle's env-dependent `:rollup_delay` runs out, the worker will call this `run` method (and also reschedule it again for another `:rollup_delay` from now)
- this method combines the old part, which is everything up to but not including today -- and which is cached for 15 days, so could end up being as many as 14 days old? -- with as many day-long mid-parts as there are between the old-part and now, and then with as many new parts as there are between mid part and now? 
- and then the combiner function puts everything together, collating where it should?

Reply:
Yess!  That's a better explanation than I've given to myself.